code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -521,10 +519,14 @@ pub mod pallet {
 			fnft_instance_id: T::FinancialNftInstanceId,
 			amount: T::Balance,
 		) -> DispatchResult {
-			let owner = ensure_signed(origin)?;
+			let who = Self::ensure_stake_owner(
```

Comment:
Would be good to verify this in a small unit test for each action

Reply:
4158f3fe0dfcdeded697191ee0559424b790da10